cylc/flow/job_runner_handlers/documentation.py

```diff
@@ -0,0 +1,412 @@
+
+        """
+        raise NotImplementedError()
+
+    def filter_submit_output(self, out: str, err: str) -> Tuple[str, str]:
+        """Filter job submission stdout/err.
+
+        Filter the standard output and standard error of the job submission
+        command. This is useful if the job submission command returns information
+        that should just be ignored.
+
+        See also :py:meth:`ExampleHandler.SUBMIT_CMD_TMPL`.
+
+        Args:
+            out: Job submit stdout.
+            err: Job submit stderr.
+
+        Returns:
+            (new_out, new_err)
+        """
+        raise NotImplementedError()
+
+    def format_directives(self, job_conf: dict) -> List[str]:
+        """Returns lines to be appended to the job script.
+
+        If relevant, this method formats the job directives for a job file, if
+        job file directives are relevant for the job runner. The argument
+        "job_conf" is a dict containing the job configuration.
+
+        Args:
+            job_conf: The Cylc configuration.
+
+        Returns:
+            lines
+
+        """
+        raise NotImplementedError()
+
+    def get_poll_many_cmd(self, job_id_list: List[str]) -> List[str]:
+        """Return a command to poll the specified jobs.
+
+        Args:
+            job_id_list: The listof job ids to poll.
+
+        Returns:
+            command e.g. ['foo', '--bar', 'baz']
+
+        """
+        raise NotImplementedError()
+
+    def get_submit_stdin(self, job_file_path: str, submit_opts: dict) -> Tuple:
+        """
+
+        Return a 2-element tuple ``(proc_stdin_arg, proc_stdin_value)``.
+
+        * Element 1 is suitable for the ``stdin=...`` argument of
+          ``subprocess.Popen`` so it can be a file handle, ``subprocess.PIPE``
+          or ``None``.
+        * Element 2 is the string content to pipe to stdin of the submit
+          command (relevant only if ``proc_stdin_arg`` is ``subprocess.PIPE``.
+
+        Args:
+            job_file_path: The path to the job file for this submission.
+            submit_opts: Job submission options.
+
+        Returns:
+            (proc_stdin_arg, proc_stdin_value)
+
+        """
+        raise NotImplementedError()
+
+    def get_vacation_signal(self, job_conf: dict) -> str:
+        """Return the vacation signal.
+
+        If relevant, return a string containing the name of the signal that
+        indicates the job has been vacated by the job runner.
+
+        Args:
+            job_conf: The Cylc configuration.
+
+        Returns:
+            signal
+
+        """
+        raise NotImplementedError()
+
+    def submit(self, job_file_path: str, submit_opts: dict) -> Tuple[int, str, str]:
+        """Submit a job.
+
+        Submit a job and return an instance of the Popen object for the
+        submission. This method is useful if the job submission requires logic
+        beyond just running a system or shell command.
+
+        See also :py:attr:`ExampleHandler.SUBMIT_CMD`.
```

Comment:
This isn't present in ExampleHandler so this reference doesn't build.

Reply:
I'm guessing this was meant to be `SUBMIT_CMD_TMPL`?